configure.ac

```diff
@@ -690,6 +690,10 @@ AS_CASE([$ocaml_cv_cc_vendor],
   warn_error_flag='-Werror'
   cc_warnings='-Wall'])
 
+OCAML_CC_HAS_COMPILE_FLAG([-Wold-style-declaration $warn_error_flag], [
```

Comment:
Musing, but is it strictly necessary to have the `$warn_error_flag`? It looked a bit strange in the `configure` log - if a C compiler accepts `-Wold-style-declaration` and silently ignores it, that's OK, surely?

Reply:
Yes, but the problem is that we're also using `$warn_error_flag` when calling the C compiler :)
So, if we're using clang, detecting `-Wold-style-declaration` without `$warn_error_flag` will "succeed":
```
configure:13664: checking whether the C compiler supports -Wold-style-declaration
configure:13684: gcc -c  -Wold-style-declaration  conftest.c >&5
warning: unknown warning option '-Wold-style-declaration'; did you mean '-Wout-of-line-declaration'? [-Wunknown-warning-option]
1 warning generated.
configure:13684: $? = 0
configure:13693: result: yes
```
 but the first invocation to it when compiling C files will fail:
```
gcc -c -O2 -fno-strict-aliasing -fwrapv -pthread -g -Wall -Wold-style-declaration -Werror -fno-common  -I./runtime  -D_FILE_OFFSET_BITS=64  -DCAMLDLLIMPORT= -DIN_CAML_RUNTIME -DNATIVE_CODE -DTARGET_amd64 -DMODEL_default -DSYS_macosx  -DCAMLDLLIMPORT= -DIN_CAML_RUNTIME  -o runtime/sak.o runtime/sak.c
error: unknown warning option '-Wold-style-declaration'; did you mean '-Wout-of-line-declaration'? [-Werror,-Wunknown-warning-option]
make: *** [runtime/sak.o] Error 1
```
(this is clang masquerading as gcc)

Reply:
Ah, I understand now, thanks! That would also mean that in releases (which don't enable `-Werror`) the `configure` output would look more logical (so opam won't get confused 😉).

I think it's worth a brief comment above the line explaining it, please!

Reply:
I've added this text:

> Clang invocation doesn't always fail if the flag is unsupported, so
> a hard error is required.